api/bin/chainflip-lp-api/src/main.rs

```diff
@@ -638,12 +658,26 @@ async fn main() -> anyhow::Result<()> {
 
 	task_scope(|scope| {
 		async move {
+			// initialize healthcheck endpoint
+			let has_completed_initialising = Arc::new(AtomicBool::new(false));
+			if opts.health_check_hostname.is_some() || opts.health_check_port.is_some() {
```

Comment:
we could also deduplicate too this by moving it into a function that wraps the healthcheck

Reply:
Also done.